lib/iris/tests/unit/common/metadata/test_BaseMetadata.py

```diff
@@ -1086,13 +1086,13 @@ def test_different(self):
             del right[key]
         expected_left, expected_right = (left, right)
         result_left, result_right = result
-        self.assertDictEqual(expected_left, result_left)
-        self.assertDictEqual(expected_right, result_right)
+        assert result_left == expected_left
+        assert result_right == expected_right
```

Comment:
Sure.

These tests were designed specifically to avoid that, as that's tested elsewhere.

Anyways, for your peace of mind I've now used the `assert_dict_equal`.